### PR TITLE
For #1214: 0crat can assign closed bug

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1214:
- Created tests assuring that `0crat` is assiging closed github issues ;
- Left puzzle for fixes.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1488](https://github.com/zerocracy/farm/issues/1488) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1488](https://github.com/zerocracy/farm/issues/1488) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2080347307); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.zerocracy.bundles.elects_and_assigns_performer
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy]: @paulodamaso wrong package name

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_before.groovy [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.zerocracy.bundles.elects_and_assigns_performer
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_before.groovy]: @paulodamaso here also wrong package name

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def exec(Project project, XML xml) {
+  String job = 'gh:test/test#1'
+  Orders orders = new Orders(project).bootstrap()
+  assert orders.performer(job) == 'yegor256'
+  Elections elections = new Elections(project).bootstrap()
+  assert !elections.elected(job)
+  Farm farm = binding.variables.farm
+  assert new LengthOf(
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy]: @paulodamaso let's migrate to `MatcherAssert`

File: src/main/groovy/com/zerocracy/stk/pm/staff/elections/elect_performer.groovy [hunk 4]
```diff
@@ -55,6 +55,11 @@ def exec(Project project, XML xml) {
   Reviews reviews = new Reviews(project).bootstrap()
   Farm farm = binding.variables.farm
   Pmo pmo = new Pmo(farm)
+  // @todo #1214:30min 0crat is assigning closed jobs. It happens when the
+  //  issue was closed in github but the Close Job flow fails
+  //  for some reason and the job does not leave WBS. Assure that we are
+  //  iterating only in open issues when electing performer and correct
+  //  _after.groovy tests in dont_assign_job_closed bundle.
```
Comment (krzyk) [inline src/main/groovy/com/zerocracy/stk/pm/staff/elections/elect_performer.groovy]: @paulodamaso I would rather explicitly write what has to be changed in the test, or do it in the test and ask for uncommenting (I think I prefer the later)

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import com.zerocracy.pm.Footprint
+import com.zerocracy.pm.in.Orders
+import com.zerocracy.pm.staff.Elections
+import org.cactoos.iterable.LengthOf
+
+
+def exec(Project project, XML xml) {
+  String job = 'gh:test/test#1'
+  Orders orders = new Orders(project).bootstrap()
+  assert orders.performer(job) == 'yegor256'
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy]: @paulodamaso `MatcherAssert`

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  Elections elections = new Elections(project).bootstrap()
+  assert !elections.elected(job)
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_after.groovy]: @paulodamaso `MatcherAssert`

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_before.groovy [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import com.jcabi.github.Repo
+import com.jcabi.github.Repos
+import com.jcabi.xml.XML
+import com.zerocracy.Farm
+import com.zerocracy.Project
+import com.zerocracy.entry.ExtGithub
+
+def exec(Project project, XML xml) {
+  Farm farm = binding.variables.farm
+  Github github = new ExtGithub(farm).value()
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/_before.groovy]: @paulodamaso github can be inlined

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/claims.xml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<claims xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://datum.zerocracy.com/0.27/xsd/pm/claims.xsd" version="0.1" updated="2017-03-27T11:18:09.228Z">
+  <claim id="1">
+    <created>2017-03-27T11:18:09.228Z</created>
+    <type>ping</type>
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/claims.xml]: @paulodamaso it should be `Ping`

File: src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/claims.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    <params>
+      <param name="force">yes</param>
+      <param name="reason">Need it</param>
+    </params>
+  </claim>
+  <claim id="2">
+    <created>2017-03-27T11:19:09.228Z</created>
+    <type>ping</type>
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/dont_assign_job_closed/claims.xml]: @paulodamaso it should be `Ping`

**Comment by paulodamaso** [comment]
@krzyk Thanks for the comments, please take a look at the fixes.

**Comment by krzyk** [comment]
@paulodamaso thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 13 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1488](https://github.com/zerocracy/farm/issues/1488) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**